atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -149,7 +149,9 @@ private boolean isInitializedInternal() {
         private void runCallback() {
             try {
                 callback.runWithRetry(txManager);
-                checkAndSetStatus(ImmutableSet.of(State.INITIALIZING), State.READY);
+                if (checkAndSetStatus(ImmutableSet.of(State.INITIALIZING), State.READY)) {
+                    executorService.shutdown();
```

Comment:
Merging first, addressing comments later 👎 : 
We can change that constructor to be package private if we are worried about this. I don't think this would hit us in practice tbh

Reply:
I think this matters more if we consider letting the environment outside atlas inject executors, since at that point we might rely on the environment to control the lifetime completely. Maybe I'm mistaken though and we only want that pattern for cases where the executor is in use for the lifetime of the application (or at least the transaction manager).